rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -219,12 +219,18 @@ class ClassicTechniqueWriter extends AgentSpecificTechniqueWriter {
   import ResultHelper._
   def writeAgentFile( technique : Technique, methods : Map[BundleName, GenericMethod] )  : Result[Option[String]] = Right(None)
   def agentMetadata ( technique : Technique, methods : Map[BundleName, GenericMethod] )  : Result[NodeSeq] = {
-    // We need to add a reporting bundle for this method to generate a na report for any method with a condition != any/cfengine (which ~= true)
+    // We need to add reporting bundle if there is a method call that does not support cfengine (agent support does not contains both cfe agent)
+    val noAgentSupportReporting = technique.methodCalls.exists(  m =>
+                         methods.get(m.methodId).exists( gm =>
+                           ! (gm.agentSupport.contains(AgentType.CfeEnterprise) || (gm.agentSupport.contains(AgentType.CfeCommunity)))
+                         )
+                       )
+    // We need to add a reporting bundle for this method to generate a na report for any method with a condition != any/cfengine (which ~= true
     val needReportingBundle = technique.methodCalls.exists(m => m.condition != "any" && m.condition != "cfengine-community" )
     val xml = <AGENT type="cfengine-community,cfengine-nova">
       <BUNDLES>
         <NAME>{technique.bundleName.value}</NAME>
-        {if (needReportingBundle) <NAME>{technique.bundleName.value}_rudder_reporting</NAME>}
+        {if (noAgentSupportReporting || needReportingBundle) <NAME>{technique.bundleName.value}_rudder_reporting</NAME>}
```

Comment:
this fails if the technique has only generic method from one type:
<pre>
Undefined bundle DSC_only_rudder_reporting with type usebundle
</pre>